shraga_common/flows/builtin/agentic_plan.py

```diff
@@ -118,6 +119,11 @@ async def execute(self, request: FlowRunRequest) -> FlowResponse:
         except (RequestCancelledException, LLMServiceUnavailableException):
             raise
         except Exception as e:
+            if not self.llm_context.is_retry:
```

Comment:
I think you could encapsulate this whoe logic inside the bedrock service @dinorn